lib/puppet/type/group.rb

```diff
@@ -99,12 +99,20 @@ def insync?(current)
       def is_to_s(currentvalue)
         if provider.respond_to?(:members_to_s)
           currentvalue = '' if currentvalue.nil?
-          return provider.members_to_s(currentvalue.split(','))
+          currentvalue = currentvalue.is_a?(Array) ? currentvalue : currentvalue.split(',')
+
+          return provider.members_to_s(currentvalue)
         end
 
         super(currentvalue)
       end
       alias :should_to_s :is_to_s
+
+      validate do |value|
+        if provider.respond_to?(:member_valid?)
+          return provider.member_valid?(value)
```

Comment:
Yeah, thanks for reminding me about that.

I was going to add a note that this approach might have a problem with a manifest that creates users at the same time.  We need to test this appropriately to make sure it's not a problem.